src/components/chip-group/chip-group.tsx

```diff
@@ -0,0 +1,245 @@
+  calciteChipListener(event: CustomEvent): void {
+    const item = event.target;
+    let updatedItems = [];
+    if (this.items.includes(item)) {
+      updatedItems = this.items.filter((el) => el !== item);
+      if (this.itemIndex(item) > 0) {
+        this.focusPrevItem(item);
+      } else if (this.itemIndex(item) === 0) {
+        this.focusNextItem(item);
+      } else {
+        this.focusFirstItem();
+      }
+    }
+    this.items = updatedItems;
+  }
+
+  @Listen("calciteInternalChipRegister")
+  registerCalciteChip(event: CustomEvent): void {
+    const item = {
+      item: event.target as HTMLCalciteChipElement,
+      parent: event.detail.parent as HTMLCalciteChipGroupElement,
+      position: (event.detail.position as number) || 0
+    };
+    if (!this.items.includes(item) && item.parent && this.el === item.parent) {
+      this.items.push(item);
+    }
+    event.stopPropagation();
+  }
+
+  @Listen("calciteInternalChipSelect")
+  updateActiveItemOnChange(event: CustomEvent): void {
+    this.calciteChipInternalSelectionChange.emit({
+      requestedChip: event.detail.requestedChip
+    });
+    const selectedItems = Array.from(this.el.getElementsByTagName("calcite-chip")).filter(
+      (el) => el.selected
+    );
+    this.calciteChipGroupChange.emit({
+      selectedItems
+    });
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private focusFirstItem() {
+    const firstItem = this.items[0];
+    this.focusElement(firstItem);
+  }
+
+  private focusLastItem() {
+    const lastItem = this.items[this.items.length - 1];
+    this.focusElement(lastItem);
+  }
+
+  private focusNextItem(el): void {
+    const index = this.itemIndex(el);
+    const nextItem = this.items[index + 1] || this.items[0];
+    this.focusElement(nextItem);
+  }
+
+  private focusPrevItem(el): void {
+    const index = this.itemIndex(el);
+    const prevItem = this.items[index - 1] || this.items[this.items.length - 1];
+    this.focusElement(prevItem);
+  }
+
+  private itemIndex(el): number {
+    return this.items.indexOf(el);
+  }
+
+  private focusElement(item) {
+    const target = item as HTMLCalciteChipElement;
+    target.setFocus();
+  }
+
+  private sortItems = (items: any[]): any[] =>
+    items.sort((a, b) => a.position - b.position).map((a) => a.item);
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  render(): VNode {
+    let role = "";
+    switch (this.selectionMode) {
+      case "single":
+      case "single-persist":
+        role = "radiogroup";
+        break;
+      case "multiple":
+        role = "checkbox";
+        break;
+    }
+
+    return (
+      <Host>
+        <div class="container" role={role}>
```

Comment:
If you are assigning a role here make sure to also allow all potentially required aria attributes and events https://developer.mozilla.org/en-US/docs/web/accessibility/aria/roles/radiogroup_role https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role

Reply:
@geospatialem could you advise on this?  (at your leisure 🏖️...)

When `chip-group` is `selection-mode="multiple`, and as a result has a role of `checkbox` - 

Should we define `aria-checked` value on the `chip-group` such as:

`mixed` (at least one but less than all child `chips` are selected), 
 `true` (all child `chips` are selected), or 
 `false` (no child `chips` are selected)

?

Reply:
@geospatialem addressed below, resolving this comment :)